crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -380,6 +394,42 @@ where
         Ok(OnForkChoiceUpdated::valid(status))
     }
 
+    /// Sets the state of the canon chain tracker based on the given forkchoice update. This should
+    /// be called before issuing a VALID forkchoice update.
+    fn update_canon_chain(&self, update: &ForkchoiceState) -> Result<(), BeaconEngineError> {
```

Comment:
perhaps these checks should happen before the `make_canonical` but we can figure this out later